File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int ori=n;
        int n1=ori;
        int sum=0;
        int prod=1;
        int sum_prod;
        while(n!=0)
        {
            sum += n%10;
            n/=10;
        }
        while(ori!=0)
        {
            prod *= ori%10;
            ori/=10;
        }
        sum_prod=sum+prod;
        if(n1%sum_prod==0)
            return true;
        else
            return false;
    }
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        ori=n
        n1=ori
        summ=0
        prod=1
        while(n!=0):
        
            summ += n%10
            n//=10
        
        while(ori!=0):
        
            prod *= ori%10
            ori//=10
        
        sum_prod=summ+prod
        if(n1%sum_prod==0):
            return True
        else:
            return False